### ISSUE TITLE
Pass parameter to modal

### ISSUE THREAD

**Issue by wvh** [bootstrap-vue/bootstrap-vue#1137]
Hello!

I'm looking for a way to pass a parameter to a modal window to indicate what to show. There doesn't seem to be a way to pass an ID either to the event call or by checking which button triggered the modal. Unless I overlooked something, one can't pass context into a modal.

My use case:
I have a modal with below a lot of recursive components. Each component has an `edit` button. Clicking the button should open the item for editing in the modal. But I can't pass the ID to the event or passively read attributes of the trigger to determine what triggered the modal, so I can't show the correct item. The nested nature of the data also makes it hard to find and address the modal directly in the Vue tree.

One other option is to insert a thousand hidden modals for each item to be edited, which doesn't seem to be a great solution performance-wise. This seems to need a dynamic and imperative approach.

Please consider an option to pass a parameter to modal windows.

**Comment by tmorehouse** [comment]
The issue with passing a parameter to the modal when opening, is that the parameter would be difficult to gt into your inner modal content, as that content is in the `this` scope of the parent, not the scope of the modal.

Your best bet is to create your own method (which accepts parameters) that programmatically opens a common modal (ie. on `@click`), and then updates the contents in the modal.

**Comment by wvh** [comment]
I understand the slot problem. It's hard to address the modal itself even because the components (that call it) are nested.

Could there be a way to pass a parameter to the `open` event then? Or a way to check which button triggered the `open` event?

It doesn't seem too uncommon a use-case to need some dynamic data inside the modal.

**Comment by mosinve** [comment]
@tmorehouse already told you the solution of your use case. You should create your custom controller component (let name it `editor`), which will handle all data&interactions, and use the modal as view component for displaying that data. This way you can get dynamic data inside modal. Take modal as View not Controller.

**Comment by ZameelAmjed** [comment]
I had to pass the id as a hidden value to the form and the method i found easy was to pass the value from button rather boolean to show modal and it worked! 
`<button @click="showModal = {{$order->id}}">Submit</button>`
and use it in template
```
 <modal v-if="showModal">
  <template v-slot:body>
    <input type="hidden" name="order_id" :value="showModal">
  </template>
</modal>
```

**Comment by wvh** [comment]
Looking back at this a few years on, while I did just start to understand the data versus imperative approach, my confusion was caused by my data being a deeply nested tree and not figuring out how to get the view (modal) pointing to the right path inside that tree. With other words, I wanted the modal to take a pointer to dynamically point inside the data, because I didn't have a fixed variable to point the modal to.

If I were to revisit this old project, I would avoid any approach of Vue components pointing to data inside deeply nested trees. If the data would have to be a tree, I would abstract away this fact in pure Javascript and just let Vue handle simple data structures.
